api/ruxos_posix_api/src/imp/execve/stack.rs

```diff
@@ -1,55 +1,56 @@
-use core::{mem::size_of, ptr::null_mut};
+use alloc::{vec, vec::Vec};
 
-use crate::*;
+use ruxconfig::TASK_STACK_SIZE;
+
+const STACK_SIZE: usize = TASK_STACK_SIZE;
 
 #[derive(Debug)]
 pub struct Stack {
-    sp: usize,
-    start: usize,
-    end: usize,
+    /// stack
+    data: Vec<u8>,
+    /// index of top byte of stack
+    top: usize,
 }
 
 impl Stack {
-    // alloc a stack
+    /// alloc a stack
     pub fn new() -> Self {
-        let size = 0xa00000; // 10M
-        let p = sys_mmap(null_mut(), size, 0, 0, 0, 0);
-
-        let start = p as usize;
-        let sp = start + size;
-        let end = sp;
+        Self {
+            data: vec![0u8; STACK_SIZE],
+            top: STACK_SIZE,
+        }
+    }
 
-        Self { sp, start, end }
+    /// panic if overflow
+    fn panic_if_of(&self) {
+        assert!(self.top <= self.data.len(), "sys_execve: stack overflow.");
     }
 
+    /// move sp to align
     pub fn align(&mut self, align: usize) -> usize {
-        self.sp -= self.sp % align;
-        self.sp
+        self.top -= self.top % align;
+        self.top
+    }
+
+    /// addr of top of stack
+    pub fn sp(&self) -> usize {
+        self.data.as_ptr() as usize + self.top
     }
 
-    pub fn push<T: Copy>(&mut self, thing: alloc::vec::Vec<T>, align: usize) -> usize {
-        let size = thing.len() * size_of::<T>();
-        self.sp -= size;
-        self.sp = self.align(align); // align 16B
+    /// push data to stack and return the addr of sp
+    pub fn push<T>(&mut self, data: &[T], align: usize) -> usize {
+        // move sp to right place
+        self.top -= core::mem::size_of_val(data);
+        self.top = self.align(align);
 
-        if self.sp < self.start {
-            panic!("stack overflow");
-        }
+        self.panic_if_of();
```

Comment:
move the defination of  `panic_if_of` there directly instead of using a function call.